src/UI/Component/Input/Field/TagInput.php

```diff
@@ -0,0 +1,134 @@
+<?php
+declare(strict_types=1);
+
+namespace ILIAS\UI\Component\Input\Field;
+
+use ILIAS\UI\Component\JavaScriptBindable;
+use ILIAS\UI\Component\Signal;
+
+/**
+ * Interface TagInput
+ *
+ * This describes Tag Inputs
+ *
+ * @package ILIAS\UI\Component\Input\Field
+ */
+interface TagInput extends Input, JavaScriptBindable {
+
+	const EVENT_ITEM_ADDED = 'itemAdded';
+	const EVENT_BEFORE_ITEM_REMOVE = 'beforeItemRemove';
+	const EVENT_BEFORE_ITEM_ADD = 'beforeItemAdd';
+	const EVENT_ITEM_REMOVED = 'itemRemoved';
```

Comment:
I think these do no belong here. As a consumer of this interface I would not know what to do with these. They look like something from the implementation to me.